### PR TITLE
rubocop: Fix commas and whitespace and newlines

### PR DESCRIPTION
**Pull request by bastelfreak**
This PR also contains https://github.com/voxpupuli/beaker/pull/1767

### CODE REVIEW

**Comment by bastelfreak** [comment]
Easier review by hiding whitespace changes: https://github.com/voxpupuli/beaker/pull/1768/files?diff=unified&w=1

**Comment by ekohl** [comment]
Can we first drop stuff? Like https://github.com/voxpupuli/beaker/pull/1769.

**Comment by bastelfreak** [comment]
this is now rebased against our master branch

File: .rubocop.yml [hunk 1]
```diff
@@ -13,23 +13,23 @@ AllCops:
...
 Layout/LineLength:
   Exclude:
     - acceptance/**/*.rb
     - spec/**/*.rb
 
-Layout:
-  Enabled: false
-
-Metrics:
-  Enabled: false
```
Comment (ekohl) [inline .rubocop.yml:28]: Can we keep Metrics disabled until a later PR? I think this is now just moving things to the TODO file and not fixing anything.

File: Rakefile [hunk 2]
```diff
@@ -121,12 +117,11 @@ def beaker_test(mode = :base, options = {})
   overriding_options = ENV['OPTIONS'].to_s
 
   args = [options_opt, hosts_opt(preserved_hosts_mode), tests_opt,
-          *overriding_options.split(' ')].compact
+          *overriding_options.split(' '),].compact
```
Comment (ekohl) [inline Rakefile:120]: This looks incorrect. Not sure how to teach RuboCop that it shouldn't be there.

File: acceptance/tests/base/dsl/helpers/host_helpers/rsync_to_test.rb [hunk 3]
```diff
@@ -12,8 +11,9 @@ def rsync_to_with_backups hosts, local_filename, remote_dir
         ) # return of block is whether or not we're done repeating
         return result.success? if result.is_a? Rsync::Result
 
-        result.each do |individual_result| 
+        result.each do |individual_result|
```
Comment (ekohl) [inline acceptance/tests/base/dsl/helpers/host_helpers/rsync_to_test.rb:14]: Or we clean it up as https://github.com/voxpupuli/beaker/pull/1776.

File: acceptance/tests/base/dsl/helpers/host_helpers/create_remote_file_test.rb [hunk 4]
```diff
@@ -13,8 +12,9 @@ def create_remote_file_with_backups host, remote_filename, contents, opts={}
           return result.success?
         end
 
-        result.each do |individual_result| 
+        result.each do |individual_result|
```
Comment (ekohl) [inline acceptance/tests/base/dsl/helpers/host_helpers/create_remote_file_test.rb:15]: https://github.com/voxpupuli/beaker/pull/1776 cleans it up in a better way

File: lib/beaker/dsl.rb [hunk 5]
```diff
@@ -1,5 +1,5 @@
-[ 'install_utils', 'roles', 'outcomes', 'assertions', 'patterns',
-  'structure', 'helpers', 'wrappers', 'test_tagging' ].each do |lib|
+['install_utils', 'roles', 'outcomes', 'assertions', 'patterns',
+ 'structure', 'helpers', 'wrappers', 'test_tagging',].each do |lib|
```
Comment (ekohl) [inline lib/beaker/dsl.rb:2]: This trailing comma also looks odd.

File: lib/beaker/shared/timed.rb [hunk 6]
```diff
@@ -1,14 +1,11 @@
 module Beaker
   module Shared
     module Timed
-
       def run_and_report_duration
         start = Time.now
```
Comment (ekohl) [inline lib/beaker/shared/timed.rb:5]: Unrelated, but this should use monotonic time.

**Review by ekohl — COMMENTED** [review]
I didn't go through all files yet, but some things that I noticed.

I'd also prefer to avoid such a huge RuboCop TODO file. Would it be better to include more fixes before we enable all layout/style cops globally?

File: .rubocop.yml (hunk 1 above)
Comment (bastelfreak) [reply .rubocop.yml:28]: done

File: Rakefile (hunk 2 above)
Comment (bastelfreak) [reply Rakefile:120]: this comes from:
```
Rakefile:120:11: C: [Corrected] Style/TrailingCommaInArrayLiteral: Put a comma after the last item of a multiline array. (https://rubystyle.guide#no-trailing-array-commas)
          *overriding_options.split(' ')].compact
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

we have this in the rubocop.yml:
```
Style/TrailingCommaInArrayLiteral:
  Enabled: True
  EnforcedStyleForMultiline: consistent_comma
```

File: acceptance/tests/base/dsl/helpers/host_helpers/rsync_to_test.rb (hunk 3 above)
Comment (bastelfreak) [reply acceptance/tests/base/dsl/helpers/host_helpers/rsync_to_test.rb]: done

File: lib/beaker/dsl.rb (hunk 5 above)
Comment (bastelfreak) [reply lib/beaker/dsl.rb]: I was able to fix this by moving it to a single line. it's 132 chars long and 140 is the current limit. As an alternative, we could disable the cop for that two lines.

File: .rubocop.yml (hunk 1 above)
Comment (bastelfreak) [reply .rubocop.yml:28]: I also disabled Style and Layout for now, to keep the file smaller.

File: lib/beaker/dsl.rb (hunk 5 above)
Comment (smortex) [reply lib/beaker/dsl.rb]: In these cases I generally put the `]` on it's own line (not sure about the rules we have for spacing, but it is not my point here):

```ruby
foo = [
        1,
        2,
        3,
      ]
```

**Review by smortex — APPROVED** [review]
My eyes could only see legit blank changes, LGTM :100: !

**Pull request merged**